src/Mod/Draft/drafttests/test_modification.py

```diff
@@ -585,21 +583,19 @@ def test_draft_to_drawing(self):
         _msg("  n_sides={}".format(prism.Polygon))
         _msg("  placement={}".format(prism.Placement))
 
-        svg_template = 'Mod/Drawing/Templates/A3_Landscape.svg'
-        template = Draft.get_param("template",
-                                   App.getResourceDir() + svg_template)
-        try:
-            page = App.ActiveDocument.addObject('Drawing::FeaturePage')
-            page.Template = template
-            _msg("  Drawing view")
-            _msg("  page={}".format(page.TypeId))
-            _msg("  template={}".format(page.Template))
-            obj = Draft.make_drawing_view(prism, page, otherProjection=None)
-            self.assertTrue(obj, "'{}' failed".format(operation))
-        except TypeError:
-            pass
-        except ModuleNotFoundError:
-            pass
+        page = App.ActiveDocument.addObject("TechDraw::DrawPage")
+        _msg("  page={}".format(page.TypeId))
+        template = App.ActiveDocument.addObject("TechDraw::DrawSVGTemplate")
+        template.Template = App.getResourceDir() \
+                            + "Mod/TechDraw/Templates/A3_Landscape_blank.svg"
+        page.Template = template
+        _msg("  template={}".format(template.TypeId))
+        view = App.ActiveDocument.addObject("TechDraw::DrawViewDraft")
+        view.Source = prism
+        view.Direction = App.Vector(0, 0, 1)
+        page.addView(view)
+        _msg("  view={}".format(view.TypeId))
+        self.assertTrue(view, "'{}' failed".format(operation))
```

Comment:
If I get it well, this doesn't ensure `page.addView(view)` went well. So you think it can be improved ?

Reply:
Sure.